unix/datetimepicker.c

```diff
@@ -532,6 +541,68 @@ static void dateTimePickerWidget_class_init(dateTimePickerWidgetClass *class)
 {
 	G_OBJECT_CLASS(class)->dispose = dateTimePickerWidget_dispose;
 	G_OBJECT_CLASS(class)->finalize = dateTimePickerWidget_finalize;
+
+	changedSignal = g_signal_new("changed",
+		G_TYPE_FROM_CLASS(class),
+		G_SIGNAL_RUN_LAST,
+		0,
+		NULL, NULL, NULL,
+		G_TYPE_NONE, 0);
+}
+
+struct uiDateTimePicker {
+	uiUnixControl c;
+	GtkWidget *widget;
+	dateTimePickerWidget *d;
+	void (*onChanged)(uiDateTimePicker *, void *);
+	void *onChangedData;
+};
+
+uiUnixControlAllDefaults(uiDateTimePicker)
+
+static void defaultOnChanged(uiDateTimePicker *d, void *data)
+{
+	// do nothing
+}
+
+void uiDateTimePickerTime(uiDateTimePicker *d, struct tm *time)
+{
+	time_t t;
+	struct tm tmbuf;
+	GDateTime *dt;
+
+	dt = selected(d->d);
+	t = g_date_time_to_unix(dt);
+	g_date_time_unref(dt);
+
+	tmbuf = *localtime(&t);
```

Comment:
I wonder if combining these two lines instead of having the `tmpbuf` would be better. `localtime()` returns a pointer already, after all...